storybookjs/storybook
# Components: Fix Code component to render children as array

Issue: #15282 #15359

The problem was caused when passing `children` as an array to the `Code` component. Since the former solution used type-cast and expected the `children` to be always string, `children.match` would throw an error for non-string `children`

## What I did
I made the Code component parsing algorithm a bit more robust to accommodate for an array.

## How to test
Pass different possible types of children to Code component:
```
<Code>{'first'}, second</Code>
<Code>{['first', 'second']}</Code>
<Code>{<b>first</b>}, second</Code>
<Code>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Duis rutrum {'commodo nibh'}, sollicitudin posuere elit varius sed.
  Curabitur non risus et eros blandit consequat.
  Cras rutrum dignissim convallis.
</Code>
```
There are all hardcoded, but there might be an expression that evaluated to string

- Is this testable with Jest or Chromatic screenshots? No
- Does this need a new example in the kitchen sink apps? No
- Does this need an update to the documentation? No

If your answer is yes to any of these, please make sure to include it in your PR.
